bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java

```diff
@@ -357,93 +353,119 @@ protected void updateStatus() {
+                if (!profile.hasBattery) { // ignore Timeouts for Sensors
+                    status = "offline.status-error-timeout";
+                }
+            } else if (res.isHttpAccessUnauthorized()) {
+                status = "offline.conf-error-access-denied";
             } else {
-                logger.debug("{}: Unable to update status: {} ({})", thingName, e.getMessage(), e.getClass());
-                status = "@text/offline.status-error-unexpected-api-result";
+                status = "offline.status-error-unexpected-api-result";
             }
-            if (!status.isEmpty() || (profile != null && !profile.isSensor)) {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, status);
+
+            if (!status.isEmpty()) {
+                setThingOffline(ThingStatusDetail.COMMUNICATION_ERROR, status);
             }
-        } catch (NullPointerException e) {
-            logger.warn("{}: Unable to update status: {} ({})", thingName, e.getMessage(), e.getClass());
+        } catch (IllegalArgumentException | NullPointerException e) {
+            logger.warn("{}: {}{}", thingName, messages.get("statusupdate.failed"), e.toString());
         } finally {
             if (scheduledUpdates > 0) {
                 --scheduledUpdates;
-                logger.debug("{}: {} more updates requested", thingName, scheduledUpdates);
-            }
-            if ((skipUpdate >= cacheCount) && !channelCache) {
+                logger.trace("{}: {} more updates requested", thingName, scheduledUpdates);
+            } else if ((skipUpdate >= cacheCount) && cache.isEnabled()) {
                 logger.debug("{}: Enabling channel cache ({} updates / {}s)", thingName, skipUpdate,
                         cacheCount * UPDATE_STATUS_INTERVAL_SECONDS);
-                channelCache = true;
+                cache.enable();
             }
         }
+    }
+
+    public boolean isThingOnline() {
+        return getThing().getStatus() == ThingStatus.ONLINE;
+    }
+
+    public boolean isThingOffline() {
+        ThingStatus status = getThing().getStatus();
+        return status == ThingStatus.OFFLINE || status == ThingStatus.UNKNOWN;
+    }
+
+    public void setThingOnline() {
+        if (!isThingOnline()) {
+            setThingStatus(ThingStatus.ONLINE);
+            requestUpdates(!profile.hasBattery ? 3 : 1, false); // request 3 updates in a row (during the first 2+3*3
+                                                                // sec)
+        }
+    }
+
+    public void setThingOffline(ThingStatusDetail detail, String messageKey) {
+        if (getThing().getStatus() != ThingStatus.OFFLINE) {
+            setThingStatus(ThingStatus.OFFLINE, detail, messageKey);
+            channelsCreated = false; // check for new channels after devices gets re-initialized (e.g. new
+        }
+    }
 
+    public void setThingStatus(ThingStatus newStatus) {
+        if (getThing().getStatus() != newStatus) {
```

Comment:
This is already handled for you by the BaseThingHandler. So you don't need this method.

Reply:
removed